ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx

```diff
@@ -0,0 +1,139 @@
+  BackgroundColor,
+  BorderColor,
+  BorderStyle,
+  Display,
+  JustifyContent,
+  TextColor,
+} from '../../../../../helpers/constants/design-system';
+import { sortAssets } from '../../util/sort';
+import { useNativeTokenBalance } from '../native-token/use-native-token-balance';
+import ImportControl from '../import-control';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+
+type AssetListControlBarProps = {
+  tokenList: TokenWithBalance[];
+  setTokenList: (arg: TokenWithBalance[]) => void;
+  showTokensLinks?: boolean;
+};
+
+const AssetListControlBar = ({
+  tokenList,
+  setTokenList,
+  showTokensLinks,
+}: AssetListControlBarProps) => {
+  const [sorted, setSorted] = useState(false);
+  const controlBarRef = useRef<HTMLDivElement>(null); // Create a ref
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const tokenSortConfig = useSelector((state: any) => {
+    return state.metamask.preferences.tokenSortConfig;
+  });
+  const selectedAccount = useSelector(getSelectedAccount);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+  const { tokensWithBalances, loading } = useAccountTotalFiatBalance(
+    selectedAccount,
+    shouldHideZeroBalanceTokens,
+  ) as {
+    tokensWithBalances: TokenWithBalance[];
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    mergedRates: any;
+    loading: boolean;
+  };
+
+  const nativeTokenWithBalance = useNativeTokenBalance();
+
+  const handleSort = () => {
+    if (!sorted) {
+      if (tokenSortConfig) {
+        const sortedTokenList = sortAssets(
+          [nativeTokenWithBalance, ...tokensWithBalances],
+          tokenSortConfig,
+        );
+        setTokenList(sortedTokenList);
+      } else {
+        setTokenList([nativeTokenWithBalance, ...tokensWithBalances]);
+      }
+    }
+  };
+
+  useEffect(() => {
+    handleSort();
+  }, [tokenSortConfig, loading, tokensWithBalances.length]);
+
+  const handleOpenPopover = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  return (
+    <>
+      <Box
+        className="asset-list-control-bar"
+        ref={controlBarRef}
+        display={Display.Flex}
+        justifyContent={JustifyContent.spaceBetween}
+        marginLeft={4}
+        marginRight={4}
+        paddingTop={4}
+      >
+        <ButtonBase
+          className="asset-list-control-bar__button"
+          onClick={handleOpenPopover}
+          size={ButtonBaseSize.Sm}
+          endIconName={IconName.ArrowDown}
+          backgroundColor={
+            isPopoverOpen
+              ? BackgroundColor.backgroundPressed
+              : BackgroundColor.backgroundDefault
+          }
+          borderColor={BorderColor.borderMuted}
+          borderStyle={BorderStyle.solid}
+          color={TextColor.textDefault}
+        >
+          Sort By
```

Comment:
This needs localization

Reply:
https://github.com/MetaMask/metamask-extension/pull/27184/commits/3107aa34e605df2c5cf5e7d73e60c2779803b669